### PR TITLE
Photo Editor sample; C++/WinRT version initially

### PR DESCRIPTION
**Pull request by stevewhims**
## Photo Editor sample

This project is the result of following the case study in [A Windows App SDK migration of the UWP Photo Editor sample app (C++/WinRT)](https://docs.microsoft.com/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/case-study-2).

This is the C++/WinRT version initially. The C# version of the same sample will follow shortly, and will go into a new `cs-winui-packaged` subfolder.

## Checklist

- [X] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [X] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release). 
- [X] Samples set the minimum supported OS version to Windows 10 version 1809.
- [X] Samples build clean with no warnings or errors.
- [X] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).

### CODE REVIEW

File: Samples/PhotoEditor/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+page_type: sample
+languages:
+- cppwinrt
+- cpp
+products:
+- windows
+- windows-app-sdk
+name: "Photo Editor"
+urlFragment: PhotoEditor
+description: "Shows how to retrieve photos from the **Pictures** library, and then edit the selected image with assorted photo effects."
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/README.md:11]:
```suggestion
description: "Shows how to retrieve photos from the Pictures library, and then edit the selected image with assorted photo effects."
```

Seems like the markdown bold format doesn't show up in the metadata format.

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl:1]: Is this file required?

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/readme.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+========================================================================
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/readme.txt:1]: Can probably delete this file, I think this comes by default when creating a new C++ WinAppSDK project.

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor.sln [hunk 4]
```diff
@@ -0,0 +1,43 @@
+﻿
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor.sln:1]: We should rename the folder `PhotoEditor/cpp-winui` - packaged is the default so the guidelines are to not include "packaged" in the folder name unless we need to distinguish the sample with an unpackaged version.

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.xaml.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "App.xaml.h"
+#include "MainWindow.xaml.h"
+
+using namespace winrt;
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.xaml.cpp:9]: For the .xaml.cpp files, the recommendation is to define the `using namespace` statments inside a `namespace winrt { }`: See the [template](https://github.com/microsoft/WindowsAppSDK-Samples/blob/67cb2ca49dc1d07bcbbc6d4f869687385b0a8d73/Templates/VSIX/Project%20Templates/cpp-winui-template/App.xaml.cpp#L9) files as an example

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/DetailPage.idl [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/DetailPage.idl:1]: Consider consolidating all `.idl` files in a `Project.idl` like in [the template file](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Templates/VSIX/Project%20Templates/cpp-winui-template/Project.idl)

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/Package.appxmanifest [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="96eb8b1a-8885-4cd2-ab4d-8f31ffc46bc7"
+    Publisher="CN=stwhi"
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/Package.appxmanifest:11]: Guidelines are to set Publisher and PublisherDisplayName to "Microsoft Corporation" - see the [template Package.appxmanifest](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Templates/VSIX/Project%20Templates/cpp-winui-template/Package.appxmanifest) as an example

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/PhotoEditor.vcxproj [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|arm64">
+      <Configuration>Release</Configuration>
+      <Platform>arm64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Configuration">
+    <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (angelazhangmsft) [inline Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/PhotoEditor.vcxproj:58]: To support building samples on both VS 2019 and VS 2022, we've been importing the Build.Common.cpp.props file in this repo instead of defining the specific version of PlatformToolset. See the [C++ windowing sample](https://github.com/microsoft/WindowsAppSDK-Samples/blob/67cb2ca49dc1d07bcbbc6d4f869687385b0a8d73/Samples/Windowing/cpp-winui/SampleApp/SampleApp.vcxproj#L6) for an example.

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl (hunk 2 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl:1]: This file is created as part of the Visual Studio project templates (for both UWP and WinAppSDK). The App class is a runtime class AFAIK, so I'd expect that it would need an IDL file.

File: Samples/PhotoEditor/README.md (hunk 1 above)
Comment (stevewhims) [reply Samples/PhotoEditor/README.md]: Fixed

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl (hunk 2 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.idl]: Fixed (deleted)

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/readme.txt (hunk 3 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/readme.txt]: Fixed (deleted)

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor.sln (hunk 4 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor.sln]: Fixed

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.xaml.cpp (hunk 5 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/App.xaml.cpp]: Fixed

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/DetailPage.idl (hunk 6 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/DetailPage.idl]: Fixed

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/Package.appxmanifest (hunk 7 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/Package.appxmanifest]: Fixed

File: Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/PhotoEditor.vcxproj (hunk 8 above)
Comment (stevewhims) [reply Samples/PhotoEditor/cpp-winui-packaged/PhotoEditor/PhotoEditor.vcxproj]: Fixed

**Review by angelazhangmsft — APPROVED** [review]
